django_project/geocontext/models/context_service_registry.py

```diff
@@ -278,32 +294,70 @@ def build_query_url(self, x, y, srid=4326):
         :return: URL to do query.
         :rtype: unicode
         """
+        bbox_string = None
         if self.query_type == ContextServiceRegistry.WFS:
             # construct bbox
             if srid != self.srid:
                 x, y = convert_coordinate(x, y, srid, self.srid)
             bbox = get_bbox(x, y)
             bbox_string = ','.join([str(i) for i in bbox])
-
             parameters = {
                 'SERVICE': 'WFS',
                 'REQUEST': 'GetFeature',
                 'VERSION': self.service_version,
                 'TYPENAME': self.layer_typename,
-                # 'SRSNAME': 'EPSG:%s' % self.srid,  # added manually
                 'OUTPUTFORMAT': 'GML3',
-                # 'BBOX': bbox_string  # added manually
             }
             query_dict = QueryDict('', mutable=True)
             query_dict.update(parameters)
-
             if '?' in self.url:
-                url = self.url + '&' + query_dict.urlencode()
+                url = '{current_url}&{urlencoded_parameters}'.format(
+                    current_url=self.url,
+                    urlencoded_parameters=query_dict.urlencode(),
+                )
             else:
-                url = self.url + '?' + query_dict.urlencode()
+                url = '{current_url}?{urlencoded_parameters}'.format(
+                    current_url=self.url,
+                    urlencoded_parameters=query_dict.urlencode(),
+                )
             # Only add SRSNAME when there is no workspace
             if ':' not in self.layer_typename:
                 url += '&SRSNAME=%s' % self.srid
             url += '&BBOX=' + bbox_string
-
+            return url
+        # ARCRest URL Construction:
+        else:
```

Comment:
There are lot of similar codes with above condition, I think we still can optimize it.